src/BloomExe/Workspace/BloomClipboard.cs

```diff
@@ -88,15 +118,29 @@ public static PalasoImage GetImageFromClipboard()
 
 			return null;
 #else
-			if (Clipboard.ContainsImage())
-			{
-				return PalasoImage.FromImage(Clipboard.GetImage());
-			}
-
 			var dataObject = Clipboard.GetDataObject();
 			if (dataObject == null)
 				return null;
 
+			var formatsAvailable = dataObject.GetFormats();
+			PalasoImage image = null;
+			var textData = String.Empty;
+			if (formatsAvailable.Contains(DataFormats.UnicodeText))
+				textData = dataObject.GetData(DataFormats.UnicodeText) as String;
+			if (Clipboard.ContainsImage())
+			{
+				image = PalasoImage.FromImage(Clipboard.GetImage());
+				if (String.IsNullOrEmpty(textData))
+				{
+					return image;
+				}
+				// we have an image on the clipboard, we also have text
+				// if the text is a valid url to an image file, use that to create a PalasoImage
+				if (File.Exists(textData))
+				{
+					return PalasoImage.FromFile(textData);
```

Comment:
Looks like this path will be attempted if the user just copies a chunk of text. What if for some reason he has copied the path to a non-image file? Feels like this needs exception handling...probably fall back to pasting the image, if any, if we can't get one from the file.